core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,225 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{
+  ScriptPubKey,
+  ScriptSignature,
+  WitnessScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{Transaction, WitnessTransaction}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import org.bitcoins.core.util.BitcoinScriptUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+object PSBTHelper {
+
+  private def parseKeyAndValue(
+      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
+    val keySize = CompactSizeUInt.parse(bytes)
+    val key = bytes.drop(keySize.size).take(keySize.toLong)
+    val valueSize =
+      CompactSizeUInt.parse(bytes.drop(keySize.size + keySize.toLong))
+    val value = bytes
+      .drop(keySize.size + keySize.toLong + valueSize.size)
+      .take(valueSize.toLong)
+    val totalSize = keySize.size + keySize.toLong + valueSize.size + valueSize
+      .toLong
+    (totalSize, key, value)
+  }
```

Comment:
I think something is wrong in this function, can't figure out what

Reply:
i can't see it either, but this seems a bit cleaner 

```scala
  private def parseKeyAndValue(
      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
    val keyCmpctUInt = CompactSizeUInt.parse(bytes)
    val key = bytes.drop(keyCmpctUInt.size).take(keyCmpctUInt.toLong)
    val valueBytes = bytes.drop(keyCmpctUInt.size + keyCmpctUInt.toLong)
    val valueCmpctUInt = CompactSizeUInt.parse(valueBytes)
    val value = valueBytes
      .drop(valueCmpctUInt.size)
      .take(valueCmpctUInt.toLong)
    val totalSize = keyCmpctUInt.size + keyCmpctUInt.toLong + valueCmpctUInt.toLong + valueCmpctUInt.size
    (totalSize, key, value)
  }
```